### PR TITLE
REACT CAPSTONE PROJECT - Air Quality Project

### PR DESCRIPTION
**Pull request by uisendy**
### Summary
This React capstone project is about building a mobile web application to check a list of metrics (numeric values) that you will create using React and Redux.

###Project Description

"AQUA" Air Quality App is an app for tracking air quality for and from any location. It connects to realtime data provided by [open weather](https://openweathermap.org/api).This app helps you to stay safe from air pollution in any country, city around the world. 

The project is a single-page application (SPA) built with React and Redux.
The data retrieved from the API should be stored in the Redux store.
You should filter the data that you retrieve from the API using a Filter stateless component.
Every page (the main page and the pages for each item) should have a unique route within the SPA.

**What you get from this app;**

1. Location Search
2. AQI - Air Quality Index for any location
3. Air Quality forcast for the next 24Hours.

**Page, Components, Fetch Data, Redux, Styling and rendering**

- The Home page which carries popular cities and the search button
- The search button routes to the search page which GETs data from the [Geocoding API](https://openweathermap.org/api/geocoding-api), which using city, country name to get location cordinates which is use to GET data from the [Air Polution API](https://openweathermap.org/api/air-pollution). 
- The details page, generated from the home page and the search page show full details about the city regarding air quality. It shows the following information 
   - The  AQI - Air Quality Index, 
   - Name of City, and Country Code. 
   - Summary report about the air Quality
   - 24 Hours forecast of the City's Air Auality Index
   - The different component in u/gm3 of the air Quality.

**Testing:** 

- Created unit tests for pure functions in your Redux code.
- Created integration tests for your application using the React Testing Library.
    - mock the access to the API, so that your tests don't send actual requests.
    - mock the connection to the Redux Store.

### CODE REVIEW

File: src/__test__/detailsPage.test.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+          <Routes>
+            <Route path="search/details" element={<DetailsPage />} />
+          </Routes>
+          ,
+        </BrowserRouter>
+      </Provider>,
+    )
+    .toJSON();
+  expect(detailsPage).toMatchSnapshot();
+});
```
Comment (medaminedev66) [inline src/__test__/detailsPage.test.js:23]: - [ ] Following the project requirements, you should use also React Testing Library to simulate user interaction. Check out this [link ](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/articles/testing_react_with_react_testing_library.md)to learn how to do that.

File: src/pages/DetailsPage.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      <div className="px-5 py-7">
+        <DetailsSummary airQuality={airQuality} status={status} error={error} summary />
+        <Forcast airQuality={airQuality} />
+        <ApComponents airQuality={airQuality} />
+      </div>
+    </div>
+  );
+};
+
+export default DetailsPage;
```
Comment (medaminedev66) [inline src/pages/DetailsPage.js:37]: - [ ] Your web app sounds good in your video presentation, but when I tested it locally I encountered issues when I want to navigate to a detail page of a location. It gives me the following error:

image

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @uisendy  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- The repo looks professional.
- Highly organized/clean code.
- Your web app is impressive.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/pages/DetailsPage.js (hunk 2 above)
Comment (uisendy) [reply src/pages/DetailsPage.js:37]: This is a error.message thrown to the UI. 

This tells us that the is a network issue from you end make cannot connect the local server to the API to fulfil a request, as thus the request was rejected and an error was throw back to the UI. 

Code Snap 2

File: src/__test__/detailsPage.test.js (hunk 1 above)
Comment (uisendy) [reply src/__test__/detailsPage.test.js:23]: I thank you @medaminedev66. It was a skip in activity. Thanks for pointing it out. I will fix that immediately.

File: README.md [hunk 3]
```diff
@@ -1,70 +1,80 @@
...
-## Available Scripts
+1. Location Search
+2. AQI - Air Quality Index for any location
+3. Air Quality forcast for the next 24Hours.
 
-In the project directory, you can run:
+## Video Presentation
 
-### `npm start`
+[Video Demo](https://www.loom.com/share/dbed31d9c34b4e94989b09b98f6cd155)
```
[the remark this answers is not in the document]
Comment (uisendy) [reply README.md:11]: Thanks @medaminedev66 : The commendation is very appreciated. 😃

File: README.md [hunk 4]
```diff
@@ -1,70 +1,80 @@
...
 
-Runs the app in the development mode.\
-Open [http://localhost:3000](http://localhost:3000) to view it in your browser.
+## Live Demo
 
-The page will reload when you make changes.\
-You may also see any lint errors in the console.
+[View Live ]()
```
[the remark this answers is not in the document]
Comment (uisendy) [reply README.md:16]: Thanks for Pointing this out. I actually felt i would need to deploy to the main branch after approval. However, I will deploy to netlify immediate. Thanks Again.

File: README.md [hunk 5]
```diff
@@ -1,70 +1,80 @@
...
+Feel free to check the [issues](https://github.com/uisendy/Meal-webApp/issues) page.
 
-This section has moved here: [https://facebook.github.io/create-react-app/docs/making-a-progressive-web-app](https://facebook.github.io/create-react-app/docs/making-a-progressive-web-app)
+## 🥇 Acknowledgement
 
-### Advanced Configuration
+Contributions, issues, and feature requests are welcome!
 
-This section has moved here: [https://facebook.github.io/create-react-app/docs/advanced-configuration](https://facebook.github.io/create-react-app/docs/advanced-configuration)
+Feel free to check the [issues](https://github.com/uisendy/Meal-webApp/issues) page.
```
[the remark this answers is not in the document]
Comment (uisendy) [reply README.md:72]: Thanks for the review and the good eye. I very much appreciate

[the remark this answers is not in the document]
Comment (uisendy) [reply README.md:72]: Done and Updated. Thanks

File: src/pages/DetailsPage.js (hunk 2 above)
Comment (uisendy) [reply src/pages/DetailsPage.js:37]: Thanks again, @medaminedev66 . I realized that the API links wasn't a secured link and thus, it was blocked. HTTP:// was replaced with HTTPS://

          const APP_ID = '05cac1b6e91db3212d68b7e40d3755c6';
          const AP_BASE_URL = 'https://api.openweathermap.org/data/2.5/air_pollution?';
          const FORCAST_AP_BASE_URL =
            'https://api.openweathermap.org/data/2.5/air_pollution/forecast?';
          const GEO_BASE_URL = 'https://api.openweathermap.org/geo/1.0/direct?';
          const limit = 5;

Again, thanks for the good eye.

File: src/__test__/detailsPage.test.js (hunk 1 above)
Comment (uisendy) [reply src/__test__/detailsPage.test.js:23]: Issue fixed, and test file is updated. Thanks @medaminedev66

File: src/pages/DetailsPage.js (hunk 2 above)
Comment (medaminedev66) [reply src/pages/DetailsPage.js:37]: Hi @uisendy, great work fixing this error 👏. Now the detail page is displayed without errors and I can see the air quality of my city 😄.

**Review by medaminedev66 — APPROVED** [review]
Hi @uisendy 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 💯 
Congratulations! 🎉 

### Highlights
- The web app works as expected and runs without errors.
- The project is deployed properly, the home and detail page works as expected.
- Your readme is descriptive.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**